pkg/cmd/server/origin/master.go

```diff
@@ -1119,3 +1131,76 @@ func WithPatternsHandler(handler http.Handler, patternHandler http.Handler, patt
 		handler.ServeHTTP(w, req)
 	})
 }
+
+// lazyServiceAccountSecretBasicAuthStore reads the service account token from a service account
+// and uses it as basic auth credentials.
+type lazyServiceAccountSecretBasicAuthStore struct {
```

Comment:
it's the only place it's used right now, I don't think of this as final because it's a really inelegant way for the master to prove auth to the registry.  Unfortunately, we have no other way to auth to the registry.